src/Features/Core/Portable/Intents/IntentDataProvider.cs

```diff
@@ -4,20 +4,28 @@
 
 using System;
 using System.Text.Json;
+using System.Text.Json.Serialization;
 using Microsoft.CodeAnalysis.ErrorReporting;
 
 namespace Microsoft.CodeAnalysis.Features.Intents
 {
     internal class IntentDataProvider
     {
-        private readonly JsonSerializerOptions _serializerOptions;
+        private static readonly Lazy<JsonSerializerOptions> s_serializerOptions = new Lazy<JsonSerializerOptions>(() =>
```

Comment:
💡 This looks like unnecessary complexity. Consider adding a `[PerformanceSensitive]` attribute and fill out the `Constraint` property so future readers know to not make this eager evaluation.